receipts/handler/handler.go

```diff
@@ -120,113 +74,71 @@ func (r *ReceiptHandler) GetCurrentReceipt() *types.EvmTxReceipt {
 
 func (r *ReceiptHandler) GetPendingTxHashList() [][]byte {
 	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+
 	hashListCopy := make([][]byte, len(r.txHashList))
 	copy(hashListCopy, r.txHashList)
-	r.mutex.RUnlock()
 	return hashListCopy
 }
 
 func (r *ReceiptHandler) Close() error {
-	switch r.v {
-	case ReceiptHandlerChain:
-	case ReceiptHandlerLevelDb:
-		err := r.leveldbReceipts.Close()
-		if err != nil {
-			return errors.Wrap(err, "closing receipt leveldb")
-		}
-	default:
-		return loomchain.ErrInvalidVersion
+	if err := r.leveldbReceipts.Close(); err != nil {
+		return errors.Wrap(err, "closing receipt leveldb")
 	}
 	return nil
 }
 
 func (r *ReceiptHandler) ClearData() error {
-	switch r.v {
-	case ReceiptHandlerChain:
-		r.chainReceipts.ClearData()
-	case ReceiptHandlerLevelDb:
-		r.leveldbReceipts.ClearData()
-	default:
-		return loomchain.ErrInvalidVersion
-	}
+	r.leveldbReceipts.ClearData()
 	return nil
 }
 
 func (r *ReceiptHandler) CommitCurrentReceipt() {
 	if r.currentReceipt != nil {
 		r.mutex.Lock()
+		defer r.mutex.Unlock()
 		r.receiptsCache = append(r.receiptsCache, r.currentReceipt)
 		r.txHashList = append(r.txHashList, r.currentReceipt.TxHash)
-		r.mutex.Unlock()
-
 		r.currentReceipt = nil
 	}
 }
 
 func (r *ReceiptHandler) DiscardCurrentReceipt() {
+	r.mutex.Lock()
+	defer r.mutex.Unlock()
 	r.currentReceipt = nil
 }
 
 func (r *ReceiptHandler) CommitBlock(state loomchain.State, height int64) error {
-	var err error
-
-	switch r.v {
-	case ReceiptHandlerChain:
-		r.mutex.RLock()
-		err = r.chainReceipts.CommitBlock(state, r.receiptsCache, uint64(height))
-		r.mutex.RUnlock()
-	case ReceiptHandlerLevelDb:
-		r.mutex.RLock()
-		err = r.leveldbReceipts.CommitBlock(state, r.receiptsCache, uint64(height))
-		r.mutex.RUnlock()
-	default:
-		err = loomchain.ErrInvalidVersion
-	}
-
 	r.mutex.Lock()
+	defer r.mutex.Unlock()
+	if err := r.leveldbReceipts.CommitBlock(state, r.receiptsCache, uint64(height)); err != nil {
+		return err
+	}
```

Comment:
This is different to how it worked before, before the hash list & receipts cache were reset regardless of whether or not `err == nil`. You think the old behavior was wrong?

Reply:
No, the behavior should be the same. Fixed